render-ws-java-client/src/main/java/org/janelia/render/client/MultiSemFlatFieldCorrectionClient.java

```diff
@@ -0,0 +1,195 @@
+import org.janelia.render.client.parameter.MultiProjectParameters;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.DoubleSummaryStatistics;
+import java.util.List;
+
+/**
+ * Apply a previously computed flat field estimate to correct shading in a Multi-SEM project.
+ * Originally developed for the wafer_53 Multi-SEM dataset.
+ *
+ * @author Michael Innerberger
+ */
+public class MultiSemFlatFieldCorrectionClient {
+
+	// make cache large enough to hold all flat field estimates for one layer
+	private static final ImageProcessorCache CACHE = new ImageProcessorCache(91L * 2000L * 1748L + 1, false, false);
+
+	private final Parameters params;
+	private final RenderDataClient renderClient;
+
+	public static class Parameters extends CommandLineParameters {
+		@ParametersDelegate
+		private final MultiProjectParameters multiProject = new MultiProjectParameters();
+		@Parameter(names = "--flatFieldLocation", description = "Location of the flat field estimates", required = true)
+		private String flatFieldLocation;
+		@Parameter(names = "--outputRoot", description = "Folder to write corrected images to", required = true)
+		private String outputFolder;
+		@Parameter(names = "--flatFieldFormat", description = "Format of the flat field estimates in printf style with placeholders for z-layer and sfov", required = true)
+		private String flatFieldFormat;
+		@Parameter(names = "--targetStackSuffix", description = "Suffix to append to the stack name for the corrected stack", required = true)
+		private String targetStackSuffix = "_corrected";
+		@Parameter(names = "--inputRoot", description = "Root folder for input data; if given, the structure under this root is replicated in the output folder")
+		private String inputRoot = null;
+		@Parameter(names = "--flatFieldConstantFromZ", description = "Maximum z-layer of flat field estimates to consider. All subsequent z-layers get corrected with the maxium, since the estimates can be bad for the last few z-layers. If not given, all z-layers are considered")
+		private Integer flatFieldConstantFromZ = Integer.MAX_VALUE;
+	}
+
+	public static void main(final String[] args) {
+		final ClientRunner clientRunner = new ClientRunner(args) {
+			@Override
+			public void runClient(final String[] args) throws IOException {
+
+				final Parameters parameters = new Parameters();
+				parameters.parse(args);
+				LOG.info("runClient: entry, parameters={}", parameters);
+
+				final MultiSemFlatFieldCorrectionClient client = new MultiSemFlatFieldCorrectionClient(parameters);
+				client.correctTiles();
+			}
+		};
+		clientRunner.run();
+	}
+
+	public MultiSemFlatFieldCorrectionClient(final Parameters parameters) {
+		this.params = parameters;
+		this.renderClient = new RenderDataClient(parameters.multiProject.baseDataUrl, parameters.multiProject.owner, parameters.multiProject.project);
+	}
+
+	public void correctTiles() throws IOException {
```

Comment:
Though not sure this is needed, to facilitate future usage within our spark pipeline setup, I'd separate this into 2 methods, pass the stack info in, and construct the render data client locally ... something like:

```
    public void correctStacks(final String baseDataUrl,
                              final List<StackId> stackIds,
                              final String targetStackSuffix) throws IOException {
        for (final StackId stackId : stackIds) {
            final String targetStackName = stackId.getStack() + targetStackSuffix;
            correctTilesForStack(baseDataUrl, stackId, targetStackName);
        }
    }

    public void correctTilesForStack(final String baseDataUrl,
                                     final StackId sourceStackId,
                                     final String targetStackName) throws IOException {

        final RenderDataClient renderClient = new RenderDataClient(baseDataUrl, sourceStackId.getOwner(), sourceStackId.getProject());

        final String sourceStackName = sourceStackId.getStack();
        final StackMetaData sourceStackMetaData = renderClient.getStackMetaData(sourceStackName);

        renderClient.setupDerivedStack(sourceStackMetaData, targetStackName);

        for (final double z : renderClient.getStackZValues(sourceStackId.getStack())) {
            ...

```